src/sentry_value.h

```diff
@@ -81,14 +81,25 @@ sentry_value_t sentry__value_clone(sentry_value_t value);
 
 /**
  * This appends `v` to the List `value`.
- * It will remove the first value of the list, is case the total number if items
- * would exceed `max`.
+ * To make this work properly as a ring buffer, the value list needs to have
+ * the ring buffer start index as the first element
+ * (e.g, 1 until max is exceeded, then it will update for each added item)
+ *
+ * It will remove the oldest value in the list, in case the total number of
+ * items would exceed `max`.
+ *
+ * The list is of size `max + 1` to store the start index.
```

Comment:
This is an exemplary implementation description and should land above the function definition of the C file because that is also where another implementer would be scratching their head regarding the layout.

Here, I would expect a high-level description that explains to callers what we expect from them and what they expect from us:
* append will exponentially grow (reallocate) the list until we reach `max`
* after reaching `max`, it will act as a ring buffer (no further list allocations or mem moves).
* what happens when `max` changes over multiple invocations, or at least what we expect call sites to do

The last point is particularly relevant because it is extraordinary that an argument should stay fixed over multiple invocations. This leaky abstraction for a particular _internal_ usage is okay, but we should highlight it.

Reply:
I've updated the description on both sides; let me know if you still feel something's missing/vague:

https://github.com/getsentry/sentry-native/pull/1060/files#diff-c47c40069b4dfa1e5cc367dcd84f2c8845913ad9dff9ff82686139981f95c5abR632-R644

https://github.com/getsentry/sentry-native/pull/1060/files#diff-975c006356fcc80792308ad2d771a20ca88700679287e4eabcfc94346c703e21R84-R89